drivers/sensor/stm32_vref/CMakeLists.txt

```
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+
+zephyr_library_sources(stm32_vref.c)
```

Shouldn't it be guarded?

```suggestion
zephyr_library_sources_ifdef(CONFIG_STM32_VREF, stm32_vref.c)
```

---

It's already guarded in `drivers/sensor/CMakeLists.txt`, I think doing that here would be redundant.